components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -196,8 +202,10 @@ protected void associateUser(String username, String userStoreDomain, String ten
             PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(usernameWithUserstoreDomain);
 
             if (!StringUtils.isEmpty(idp) && !StringUtils.isEmpty(subject)) {
-                UserProfileAdmin userProfileAdmin = UserProfileAdmin.getInstance();
-                userProfileAdmin.associateID(idp, subject);
+                FederatedAssociationManager federatedAssociationManager = FrameworkUtils
+                        .getFederatedAssociationManager();
+                User user = getAssociatedUser(username, userStoreDomain, tenantDomain);
```

Comment:
```suggestion
                User user = getAssociatedUser(username, userStoreDomain, tenantDomain);
```
User user = getAssociatedUser(username, tenantDomain, userStoreDomain);

If possible, can we interchange method parameters to the order -> `tenantDomain, userStoreDomain, username`